src/cogent3/core/new_sequence.py

```diff
@@ -327,84 +329,80 @@ def strip_degenerate(self):
         """Removes degenerate bases by stripping them out of the sequence."""
         return self.__class__(
             moltype=self.moltype,
-            seq=self.moltype.strip_degenerate(self),
+            seq=self.moltype.strip_degenerate(str(self)),
             info=self.info,
         )
 
     def strip_bad(self):
         """Removes any symbols not in the alphabet."""
+        # refactor: design
+        # previous implementation did NOT strip X from DNA even though its not in the alphabet
+        # do we want to keep that behavior?
         return self.__class__(
-            moltype=self.moltype, seq=self.moltype.strip_bad(self), info=self.info
+            moltype=self.moltype, seq=self.moltype.strip_bad(str(self)), info=self.info
         )
 
     def strip_bad_and_gaps(self):
         """Removes any symbols not in the alphabet, and any gaps."""
         return self.__class__(
             moltype=self.moltype,
-            seq=self.moltype.strip_bad_and_gaps(self),
+            seq=self.moltype.strip_bad_and_gaps(str(self)),
             info=self.info,
         )
 
-    def is_gapped(self):
+    def is_gapped(self) -> bool:
         """Returns True if sequence contains gaps."""
-        return self.moltype.is_gapped(self)
+        return self.moltype.is_gapped(str(self))
 
-    def is_gap(self, char=None):
+    def is_gap(self, char: str = None) -> bool:
```

Comment:
checking how often this method used, I suspect we can just delete this method and associated tests